Reduce input sizes or specify in OpInfo gradcheck_fast_mode=True

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #80515
* #80559
* #80514

~Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:~
- ~UnaryUfunc's default samples inputs~
- ~addbmm~
- ~baddbmm~
- ~matmul~
- ~atleast{1d,2d,3d}~

Reduces the sizes of the inputs for:
- diff
- diag_embed

Marks the OpInfos for the op as `fast_gradcheck` only (note that this applies at the op level, i.e. it applies to all dtypes and devices and across all TestGradient tests) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```

Update:

Additionally runs the following slow tests in fast gradcheck only
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
and
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Skips the following new failures (https://github.com/pytorch/pytorch/issues/80411)

```
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

soulitzer committed:
Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests

[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests"




[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests"




[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests"




[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests"




[ghstack-poisoned]

soulitzer committed:
Update on "[Testing only] Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests"




[ghstack-poisoned]

soulitzer committed:
Update on "Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests or mark as gradcheck_fast_mod"


Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:
- UnaryUfunc's default samples inputs 
- addbmm
- baddbmm
- diff
- diag_embed
- matmul
- atleast{1d,2d,3d}

Marks the following as `fast_gradcheck` only (applies to all dtypes and devices) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```
TODO: Think about a better way of skipping?

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

[ghstack-poisoned]

soulitzer committed:
Update on "Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests or mark as gradcheck_fast_mod"


Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:
- UnaryUfunc's default samples inputs 
- addbmm
- baddbmm
- diff
- diag_embed
- matmul
- atleast{1d,2d,3d}

Marks the OpInfos for the op as `fast_gradcheck` only (note that this applies at the op level, i.e. it applies to all dtypes and devices and across all TestGradient tests) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```

Update:

Additionally skips the following slow tests 
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```

Skips the following new failures (https://github.com/pytorch/pytorch/issues/80411)

```
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

TODO: Think about a better way of skipping?

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

[ghstack-poisoned]

mruberry commented on test/test_ops_gradients.py:
```diff
@@ -44,7 +45,10 @@ def is_inplace(variant):
 
         include_conjugated_inputs = op.test_conjugated_samples and dtype.is_complex
-        samples = op.sample_inputs(device, dtype, requires_grad=True, include_conjugated_inputs=include_conjugated_inputs)
+        small_inputs_only = ((check == "bwgrad_bwgrad" or check == "fwgrad_bwgrad") and
```
Instead of adding a new kwarg what about just shrinking the sample inputs?

mruberry replied:
That sounds fine, too

soulitzer replied:
@mruberry  Looks like it was timing out again because I removed the reduced input size changes for unary ufuncs. I added it back along with reduced input size changes for binary ufuncs as well (see latest update). Alternatively, what do you think about always skipping slow gradcheck tests for unary/binary ufuncs actually because this should be a case fast gradcheck will always catch.

soulitzer committed:
Update on "Reduce input sizes for slowest gradgrad and fwgrad_bwgrad tests or mark as gradcheck_fast_mod"


Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:
- UnaryUfunc's default samples inputs 
- addbmm
- baddbmm
- diff
- diag_embed
- matmul
- atleast{1d,2d,3d}

Marks the OpInfos for the op as `fast_gradcheck` only (note that this applies at the op level, i.e. it applies to all dtypes and devices and across all TestGradient tests) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```

Update:

Additionally runs the following slow tests in fast gradcheck only
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
and
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Skips the following new failures (https://github.com/pytorch/pytorch/issues/80411)

```
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

[ghstack-poisoned]

soulitzer committed:
Update on "Reduce input sizes or specify in OpInfo gradcheck_fast_mode=True"


~Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:~
- ~UnaryUfunc's default samples inputs~
- ~addbmm~
- ~baddbmm~
- ~matmul~
- ~atleast{1d,2d,3d}~

Reduces the sizes of the inputs for:
- diff
- diag_embed

Marks the OpInfos for the op as `fast_gradcheck` only (note that this applies at the op level, i.e. it applies to all dtypes and devices and across all TestGradient tests) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```

Update:

Additionally runs the following slow tests in fast gradcheck only
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
and
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Skips the following new failures (https://github.com/pytorch/pytorch/issues/80411)

```
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

[ghstack-poisoned]

soulitzer committed:
Update on "Reduce input sizes or specify in OpInfo gradcheck_fast_mode=True"


~Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:~
- ~UnaryUfunc's default samples inputs~
- ~addbmm~
- ~baddbmm~
- ~matmul~
- ~atleast{1d,2d,3d}~

Reduces the sizes of the inputs for:
- diff
- diag_embed

Marks the OpInfos for the op as `fast_gradcheck` only (note that this applies at the op level, i.e. it applies to all dtypes and devices and across all TestGradient tests) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```

Update:

Additionally runs the following slow tests in fast gradcheck only
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
and
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Skips the following new failures (https://github.com/pytorch/pytorch/issues/80411)

```
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

[ghstack-poisoned]

soulitzer committed:
Update on "Reduce input sizes or specify in OpInfo gradcheck_fast_mode=True"


~Reduces the sizes of the inputs for `test_fn_fwgrad_bwgrad` and `test_fn_gradgrad` in slow gradcheck for:~
- ~UnaryUfunc's default samples inputs~
- ~addbmm~
- ~baddbmm~
- ~matmul~
- ~atleast{1d,2d,3d}~

Reduces the sizes of the inputs for:
- diff
- diag_embed

Marks the OpInfos for the op as `fast_gradcheck` only (note that this applies at the op level, i.e. it applies to all dtypes and devices and across all TestGradient tests) should reduce slow gradcheck runtime by >60% (the top 30 tests (out of ~1000) comprise of about 60% of its runtime, even after applying the above changes):
```
0  918.722  test_fn_fwgrad_bwgrad_nn_functional_conv_transpose3d_cuda_float64 
1  795.042  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_complex128 
2  583.63  test_fn_fwgrad_bwgrad_nn_functional_max_pool3d_cuda_float64 
3  516.946  test_fn_fwgrad_bwgrad_svd_cuda_complex128 
4  503.179  test_fn_fwgrad_bwgrad_linalg_svd_cuda_complex128 
5  460.985  test_fn_fwgrad_bwgrad_linalg_lu_cuda_complex128 
6  401.04  test_fn_fwgrad_bwgrad_linalg_lstsq_grad_oriented_cuda_complex128 
7  353.671  test_fn_fwgrad_bwgrad_nn_functional_max_pool2d_cuda_float64 
8  321.903  test_fn_fwgrad_bwgrad_nn_functional_gaussian_nll_loss_cuda_float64 
9  307.951  test_fn_fwgrad_bwgrad_stft_cuda_complex128 
10  266.104  test_fn_fwgrad_bwgrad_svd_lowrank_cuda_float64 
11  221.032  test_fn_fwgrad_bwgrad_istft_cuda_complex128 
12  183.741  test_fn_fwgrad_bwgrad_lu_unpack_cuda_complex128 
13  132.019  test_fn_fwgrad_bwgrad_nn_functional_unfold_cuda_float64 
14  125.343  test_fn_fwgrad_bwgrad_nn_functional_pad_constant_cuda_complex128 
15  124.2  test_fn_fwgrad_bwgrad_kron_cuda_complex128 
16  123.721  test_fn_fwgrad_bwgrad_pca_lowrank_cuda_float64 
17  121.074  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
18  119.387  test_fn_fwgrad_bwgrad_rot90_cuda_complex128 
19  112.889  test_fn_fwgrad_bwgrad__masked_normalize_cuda_complex128 
20  107.541  test_fn_fwgrad_bwgrad_dist_cuda_complex128 
21  106.727  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
22  104.588  test_fn_fwgrad_bwgrad__masked_cumprod_cuda_complex128 
23  100.135  test_fn_fwgrad_bwgrad_nn_functional_feature_alpha_dropout_with_train_cuda_float64 
24  88.359  test_fn_fwgrad_bwgrad_mH_cuda_complex128 
25  86.214  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
26  83.037  test_fn_fwgrad_bwgrad_nn_functional_bilinear_cuda_float64 
27  79.987  test_fn_fwgrad_bwgrad__masked_cumsum_cuda_complex128 
28  77.822  test_fn_fwgrad_bwgrad_diag_embed_cuda_complex128 
29  76.256  test_fn_fwgrad_bwgrad_mT_cuda_complex128 
30  74.039  test_fn_fwgrad_bwgrad_linalg_lu_solve_cuda_complex128 
```

Update:

Additionally runs the following slow tests in fast gradcheck only
```
0  334.142  test_fn_fwgrad_bwgrad_unfold_cuda_complex128 
1  312.791  test_fn_fwgrad_bwgrad_linalg_lu_factor_cuda_complex128 
2  121.963  test_fn_fwgrad_bwgrad_nn_functional_max_unpool3d_cuda_float64 
3  108.085  test_fn_fwgrad_bwgrad_diff_cuda_complex128 
4  89.418  test_fn_fwgrad_bwgrad_nn_functional_max_unpool2d_cuda_float64 
5  72.231  test_fn_fwgrad_bwgrad___rdiv___cuda_complex128 
6  69.433  test_fn_fwgrad_bwgrad___getitem___cuda_complex128 
7  68.582  test_fn_fwgrad_bwgrad_ldexp_cuda_complex128 
8  68.572  test_fn_fwgrad_bwgrad_linalg_pinv_cuda_complex128 
9  67.585  test_fn_fwgrad_bwgrad_nn_functional_glu_cuda_float64 
10  66.567  test_fn_fwgrad_bwgrad_lu_cuda_float64 
```
and
```
0  630.13  test_fn_gradgrad_nn_functional_conv2d_cuda_complex128 
1  81.086  test_fn_gradgrad_linalg_solve_triangular_cuda_complex128 
2  71.332  test_fn_gradgrad_norm_cuda_complex128 
3  64.308  test_fn_gradgrad__masked_std_cuda_complex128 
4  59.519  test_fn_gradgrad_div_no_rounding_mode_cuda_complex128 
5  58.836  test_fn_gradgrad_nn_functional_adaptive_avg_pool3
```

Skips the following new failures (https://github.com/pytorch/pytorch/issues/80411)

```
test_fn_fwgrad_bwgrad_zero__cuda_complex128
test_fn_gradgrad_linalg_lu_factor_cuda_float64
test_fn_grad_div_trunc_rounding_cuda_float64
test_fn_grad_div_floor_rounding_cuda_float64
```

Follow ups after this stack has landed:
1. investigate failures
2. see if we can reenable slow gradcheck tests for some of these by reducing the sizes of their inputs

[ghstack-poisoned]
